### PR TITLE
Handle undefined in Cloud Code

### PR DESCRIPTION
**Pull request by dplewis**
Closes https://github.com/parse-community/Parse-SDK-JS/issues/417

Based on discussion https://github.com/parse-community/parse-server/issues/5053

Previously threw Invalid response and unhandled promise rejection

### CODE REVIEW

File: src/Cloud.js [hunk 1]
```diff
@@ -126,10 +125,7 @@ const DefaultController = {
       if (decoded && decoded.hasOwnProperty('result')) {
         return Promise.resolve(decoded.result);
       }
-      throw new ParseError(
-        ParseError.INVALID_JSON,
-        'The server returned an invalid response.'
-      );
```
[the remark this answers is not in the document]
Comment (dplewis) [reply src/Cloud.js]: I don’t think that would ever happen

[the remark this answers is not in the document]
Comment (flovilmart) [reply src/Cloud.js]: Ok then!

File: src/Cloud.js [hunk 2]
```diff
@@ -126,10 +125,7 @@ const DefaultController = {
...
+      return Promise.resolve(null);
```
[the remark this answers is not in the document]
Comment (dplewis) [reply src/Cloud.js]: .decode handles that internally

File: src/__tests__/Cloud-test.js [hunk 3]
```diff
@@ -151,13 +151,12 @@ describe('CloudController', () => {
 
   it('run invalid response', (done) => {
     const request = jest.fn();
-    request.mockReturnValue(Promise.resolve({
-      success: false
-    }));
+    request.mockReturnValue(Promise.resolve(undefined));
```
[the remark this answers is not in the document]
Comment (dplewis) [reply src/__tests__/Cloud-test.js]: I change it, need for unit test

File: src/Cloud.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (flovilmart) [reply src/Cloud.js]: So let’s just return the decoded result, defined or undefined. Perhaps we could check that only the result key is at least définined and no other

File: src/__tests__/Cloud-test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (flovilmart) [reply src/__tests__/Cloud-test.js]: I think the test should stay the same. Returning an object with unknown key should not be valid. 
Ie: if the response has not .result and there is at least a key, then we fail

File: src/Cloud.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dplewis) [reply src/Cloud.js]: I though about that as well. But let say we wanted to add additional response to cloud code like status OK or something

File: src/__tests__/Cloud-test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (dplewis) [reply src/__tests__/Cloud-test.js]: Oh I see what you mean

File: src/Cloud.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (flovilmart) [reply src/Cloud.js]: Then we will revisit, so far we have this failing because undefined is not serialized where all other platforms support it.

**Pull request merged**